sdk/monitor/monitor-opentelemetry/test/internal/unit/main.test.ts

```diff
@@ -251,4 +251,71 @@ describe("Main functions", () => {
     useAzureMonitor(config);
     assert.strictEqual(process.env["AZURE_MONITOR_PREFIX"], `k${os}m_`);
   });
+
+  it("should prioritize resource detectors in env var OTEL_NODE_RESOURCE_DETECTORS", () => {
+    const expectedResourceAttributeNamespaces = new Set(["os", "service", "telemetry"]);
+    const env = <{ [id: string]: string }>{};
+    env.OTEL_NODE_RESOURCE_DETECTORS = "os";
+    process.env = env;
+    let config: AzureMonitorOpenTelemetryOptions = {
+      azureMonitorExporterOptions: {
+        connectionString: "InstrumentationKey=00000000-0000-0000-0000-000000000000",
+      },
+    };
+    useAzureMonitor(config);
+    let span = trace.getTracer("testTracer").startSpan("testSpan");
+    span.end();
+
+    // @ts-ignore Need to access resource attributes of a span to verify the correct resource detectors are enabled.
+    // The resource field of a span is a readonly IResource and does not have a getter for the underlying Resource.
+    const resource = span["resource"].attributes;
+    console.log(resource);
+    Object.keys(resource).forEach((attr) => {
+      const parts = attr.split(".");
+      assert.ok(expectedResourceAttributeNamespaces.has(parts[0]));
+    });
+  });
+
+  it("should skip unknown resource detectors", () => {
+    const expectedResourceAttributeNamespaces = new Set(["host", "service", "telemetry"]);
+    const env = <{ [id: string]: string }>{};
+    env.OTEL_NODE_RESOURCE_DETECTORS = "blah,host";
+    process.env = env;
+    let config: AzureMonitorOpenTelemetryOptions = {
+      azureMonitorExporterOptions: {
+        connectionString: "InstrumentationKey=00000000-0000-0000-0000-000000000000",
+      },
+    };
+    useAzureMonitor(config);
+    let span = trace.getTracer("testTracer").startSpan("testSpan");
+    span.end();
+
+    // @ts-ignore Need to access resource attributes of a span to verify the correct resource detectors are enabled.
+    // The resource field of a span is a readonly IResource and does not have a getter for the underlying Resource.
+    const resource = span["resource"].attributes;
```

Comment:
Looked into this a bit more and it looks like you could so something like we have in the `azureFnHook.test.ts` file on line 88: `(span as any)["_spanContext"]["traceId"]`. It loses any real type meaning since it's just casting to `any`, but in my opinion it's a little cleaner than having the `@ts-ignore`. Comment is great though! Thanks for adding!